server/src/main/java/com/linecorp/centraldogma/server/internal/mirror/GitMirror.java

```diff
@@ -347,7 +347,15 @@ private Git openGit(File workDir) throws Exception {
                 CONSECUTIVE_UNDERSCORES.matcher(DISALLOWED_CHARS.matcher(
                         remoteRepoUri().toASCIIString()).replaceAll("_")).replaceAll("_"));
 
-        final GitWithAuth git = new GitWithAuth(this, repoDir);
+        GitWithAuth git;
+        try {
+            final Class<?> jgit6 = Class.forName("com.linecorp.centraldogma.server" +
+                                                 ".internal.mirror.GitWithAuth6");
```

Comment:
We can check the existence of `GitWithAuth6` in the `static` block so that we can use `GitWithAuth5` without relying on an exception.
```java
@Nullable
static Class<?> GIT_WITH_AUTH6;

static {
    try {
       GIT_WITH_AUTH6 = Class.forName("...");
    } ...
}

Reply:
Hello, Thanks for suggestion, but how do we use `GitWithAuth5` without relying on an exception? Because `Class.forName` throw an exception so I guess we always need to use catch, does it?

Reply:
You can check if `GIT_WITH_AUTH6` is `null`. Assign `null` to `GIT_WITH_AUTH6` if `GitWithAuth6` is unavailable.